RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc

```diff
@@ -401,6 +410,33 @@ void ReducedEGProducer::produce(edm::Event& theEvent, const edm::EventSetup& the
     const reco::GsfElectronCoreRef &gsfElectronCore = gsfElectron.core();
     linkCore(gsfElectronCore, *gsfElectronCores, gsfElectronCoreMap);
 
+
+    const reco::GsfTrackRef &gsfTrack = gsfElectron.gsfTrack();
+    
+    // Save the main gsfTrack
+    if (!gsfTrackMap.count(gsfTrack)) {
+      gsfTracks->push_back(*gsfTrack);
+      gsfTrackMap[gsfTrack] = gsfTracks->size() - 1;
+      //std::cout << "push back gsftrack to gsftrack ref" << std::endl;
+      //std::cout << " gsf track size: " << gsfTracks->size() << std::endl;
+    }
+    
+    // Save additional ambiguous gsf tracks in a map:
+    for (reco::GsfTrackRefVector::const_iterator igsf = gsfElectron.ambiguousGsfTracksBegin(); igsf != gsfElectron.ambiguousGsfTracksEnd(); ++igsf) {
```

Comment:
I don't want to change the loops in this PR. I based my code on the what was already there. If this must be changed, it's better to do this in a separate PR for the whole code, not just the part I committed.